A Better Issue Template (#5380)
microsoft/msbuild: .github/issue_template/01_bugreport.md

---
name: 🐞 Bug Report
about: Report a bug to help us improve MSBuild.
title: ''

> labels:

Reviewer: Can we use labels, if people don't usually have permissions to add labels? we have a `bug` label.

Reviewer: Though if we start doing that, we'll have to change our triage policy away from "has no label". That's fine!

Author: Do we want to create some sort of label for the "my build is broken" template to assign? Maybe we already have one?

---

<!-- This is a template that helps us provide quicker feedback. Please use any relevant sections and delete anything you don't need. -->

### Issue Description
<!--
* Please include a clear and concise description of the problem.
-->

### Expected Behavior
<!--
* The expected output or behavior.
-->

### Actual Behavior
<!--
* The actual output or behavior.
-->

### Analysis
<!--
* If you have an idea where the problem might lie, let us know that here.
* Please include any pointers to code, relevant changes, or related issues you know of.
-->

### Versions & Configurations
<!--
* In a Visual Studio developer command prompt, run `msbuild -version` and paste the output here.
* If applicable, include the version of the tool that invokes MSBuild (Visual Studio, dotnet CLI, etc):

* Post any other relevant configuration settings here.
*   OS, architecture, etc.
-->

> ### Steps to Reproduce

Reviewer: This section should be above expected/actual IMO

<!--
* Include as much of the following if possible:

* A minimal sample project that reproduces the issue.
* Your zipped project.
* IDE / CLI steps to create the project and reproduce the behaviour.
* Your command line invocation
-->

### Attach a binlog
<!--
* If providing us a project that reproduces the issue proves difficult, consider including a binlog.

> * Click [here](https://github.com/microsoft/msbuild/blob/master/documentation/wiki/Providing-Binary-Logs.md) for details on sharing binary logs.

Reviewer: ```suggestion
* Click [here](https://aka.ms/msbuild/binlog) for details on sharing binary logs.
```

future-proofing against eventually moving this to docs.

### Issue Description
<!--
Please include a clear and concise description of the problem.
-->

### Expected Behavior
<!--
* Include the expected output or behavior.
-->

### Actual Behavior
<!--
* Include the actual output or behavior.
-->

### Analysis
<!--
* If you have an idea where the problem might lie, let us know that here.
* Please include any pointers to code, relevant changes, or related issues you know of.
* If you don't know, you can delete this section.
-->

### Versions & Configurations
<!--
* In a visual studio developer command prompt, run `msbuild -version` and paste the output here.
* If applicable, version of the tool that invokes MSBuild (Visual Studio, dotnet CLI, etc):

Post any other relevant configuration settings here.
* OS, architecture, etc.
-->

### Steps to Reproduce
<!--
Include as much of the following if possible:

* A minimal sample project that reproduces the issue.
* Your zipped project.
* IDE / CLI steps to create the project and reproduce the behaviour.
* Your command line invocation
-->

### Attach a binlog
<!--

> To help us figure out and fix the issue sooner, please include a binlog.

Reviewer 2: ```suggestion
If it is difficult to create a simple project that reproduces the issue, consider including a binlog.
```
?

> * Follow the link for details on sharing binary logs: https://github.com/microsoft/msbuild/blob/master/documentation/wiki/Providing-Binary-Logs.md
> * Follow this link for more information on binary logs: https://github.com/microsoft/msbuild/blob/master/documentation/wiki/Binary-Log.md

Reviewer 2: ```suggestion
* Click [here](https://github.com/microsoft/msbuild/blob/master/documentation/wiki/Providing-Binary-Logs.md) for details on sharing binary logs.
* Click [here](https://github.com/microsoft/msbuild/blob/master/documentation/wiki/Binary-Log.md) for more information on binary logs.
```

    NOTE: Binlogs can contain sensitive information. Don't attach anything you don't want to be public.

> *   To view these binlogs yourself: https://msbuildlog.com/

Reviewer 2: ```suggestion
*   To view these binlogs yourself, see https://msbuildlog.com/
```

Reviewer: ```suggestion
*   To view the contents of the binlogs yourself, you may wish to use a tool like https://msbuildlog.com/.
```
